ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java

```diff
@@ -0,0 +1,73 @@
+package org.nervos.ckb.utils.address;
+
+import org.nervos.ckb.address.Network;
+import org.nervos.ckb.exceptions.AddressFormatException;
+import org.nervos.ckb.type.Script;
+import org.nervos.ckb.utils.Bech32;
+import org.nervos.ckb.utils.Numeric;
+
+/** Copyright © 2019 Nervos Foundation. All rights reserved. */
+public class AddressParser extends AddressBaseOperator {
+
+  private static String parsePayload(String address) {
+    Bech32.Bech32Data parsed = Bech32.decode(address);
+    byte[] data = convertBits(com.google.common.primitives.Bytes.asList(parsed.data), 5, 8, false);
+    if (data.length == 0) {
+      return null;
+    }
+    Bech32.Bech32Data bech32Data = new Bech32.Bech32Data(parsed.hrp, data);
+    return Numeric.toHexStringNoPrefix(bech32Data.data);
+  }
+
+  public static AddressParseResult parse(String address) throws AddressFormatException {
+    String payload = parsePayload(address);
+    if (payload == null) {
+      throw new AddressFormatException("Address bech32 decode fail");
+    }
+    String type = payload.substring(0, 2);
+    if (TYPE_SHORT.equals(type)) {
+      String codeHashIndex = payload.substring(2, 4);
+      String args = Numeric.prependHexPrefix(payload.substring(4));
+      if (Numeric.cleanHexPrefix(args).length() != 40) {
+        throw new AddressFormatException("Short address args byte length must be equal to 20");
```

Comment:
if statement check hex string length but error message show args legal length, would it be better to unify it here?